// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
import "google/protobuf/empty.proto";

option objc_class_prefix = "RTG";


// Interface exported by the server.
service TaskManager {
  rpc RunTask(TaskStatement) returns (Response) {}

  rpc KillTask(Task) returns (Response) {}

  rpc GetTaskStatus(Task) returns (Response) {}

  rpc GetAllTasks(google.protobuf.Empty) returns (AllTasksStatus) {}
}

message TaskStatement {
    int32 gpuidx = 1;
    string command = 2;
    string name = 3;
}

message Response {
    string task_id = 1;
    enum ResponseStatus {
        READY = 0;
        RUNNING = 1;
        DONE = 2;
        KILLED = 3;
        ABNORMAL = 4;
    }
    ResponseStatus response = 2; // response가 의미하는 값은 Enum으로 정의
}

message Task {
    string task_id = 1;
}

message AllTasksStatus {
    repeated Response response_array = 1;
}